src/ASM/FiniteElement.C

```diff
@@ -42,7 +48,7 @@ std::ostream& FiniteElement::write (std::ostream& os) const
 std::ostream& MxFiniteElement::write (std::ostream& os) const
 {
   this->FiniteElement::write(os);
-  for (size_t b = 0; b+1 < Nx.size(); b++)
+  for (size_t b = 0; b < Nx.size(); b++)
   {
```

Comment:
first basis is not in Nx. Nx is only the additional bases (see MxFiniteElement constructor). it's this way due to inheritance stuff (we want to look like a FiniteElement for the first basis).